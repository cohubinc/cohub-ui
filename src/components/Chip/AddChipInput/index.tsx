import React, { CSSProperties } from "react";

import { size } from "src/helpers/style-utils";
import Icon from "src/components/Icon";

const transition = "0.2s ease-in-out";
const btnInputStyle = {
  ...size(26),
  outline: "none",
  borderRadius: "50%",
  padding: 10,
  transition
};

const expandedInputStyles = {
  outline: "none",
  height: 28,
  width: 120,
  borderRadius: 32,
  padding: "3px 9px",
  transition
};

const defaultState = {
  expanded: false
};
interface IProps {
  name?: string;
  onBlur?: any;
  onChange?: any;
  onFocus?: any;
  className?: string;
  style?: CSSProperties;
}
export default class AddChipInput extends React.Component<
  IProps,
  typeof defaultState
> {
  _input: React.RefObject<HTMLInputElement>;
  state = defaultState;

  constructor(props: IProps) {
    super(props);
    this._input = React.createRef();
  }

  render() {
    const { className, style = {}, ...restProps } = this.props;

    const { expanded } = this.state;

    const defaultIconStyle = {
      position: "absolute",
      cursor: "pointer",
      height: 13,
      transition
    };
    const iconStyles = expanded
      ? {
          ...defaultIconStyle,
          left: 49,
          top: 6,
          transform: "translate(50px, 0) rotate(45deg)"
        }
      : { ...defaultIconStyle, left: 5.3, top: 5.4 };

    const inputRef = this._input.current;

    if (inputRef) {
      expanded ? inputRef.focus() : inputRef.blur();
    }

    const { onBlur, onFocus, ...inputProps } = restProps;
    return (
      <div style={{ ...style, position: "relative" }} className={className}>
        <input
          {...inputProps}
          className="border"
          style={expanded ? expandedInputStyles : btnInputStyle}
          ref={this._input}
          onFocus={e => {
            onFocus && onFocus(e);
            this.setState({ expanded: true });
          }}
          onBlur={e => {
            onBlur && onBlur(e);
            this.setState({ expanded: false });
          }}
        />

        <Icon.Add
          size={16.5}
          onClick={this.toggleState}
          style={iconStyles as any}
        />
      </div>
    );
  }

  toggleState = () => {
    this.setState(({ expanded }) => ({ expanded: !expanded }));
  };
}
