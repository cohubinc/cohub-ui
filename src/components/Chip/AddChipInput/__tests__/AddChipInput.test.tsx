import React from "react";
import { shallow } from "enzyme";

import AddChipInput from "../index";
it("renders AddChipInput without crashing", async () => {
  const wrapper = shallow(<AddChipInput />);

  expect(wrapper).toMatchSnapshot();
});
