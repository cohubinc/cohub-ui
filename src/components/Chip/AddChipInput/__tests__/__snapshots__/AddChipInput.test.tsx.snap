// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AddChipInput without crashing 1`] = `
<div
  style={
    Object {
      "position": "relative",
    }
  }
>
  <input
    className="border"
    onBlur={[Function]}
    onFocus={[Function]}
    style={
      Object {
        "borderRadius": "50%",
        "height": 26,
        "outline": "none",
        "padding": 10,
        "transition": "0.2s ease-in-out",
        "width": 26,
      }
    }
  />
  <Component
    onClick={[Function]}
    size={16.5}
    style={
      Object {
        "cursor": "pointer",
        "height": 13,
        "left": 5.3,
        "position": "absolute",
        "top": 5.4,
        "transition": "0.2s ease-in-out",
      }
    }
  />
</div>
`;
